src/quicklist.c

```diff
@@ -48,6 +48,20 @@
 /* Optimization levels for size-based filling */
 static const size_t optimization_level[] = {4096, 8192, 16384, 32768, 65536};
 
+/* packed_threshold is initialized to 1gb*/
+static size_t packed_threshold = (1 << 30);
+
+/* set threshold for PLAIN nodes, the real limit is 4gb */
+#define isLargeElement(size) (size >= packed_threshold)
+
+int quicklistisSetPackedThreshold(size_t sz) {
+    if (sz > 4294967296) {
```

Comment:
Why make this judgment? Aren't we trying to get it to support more than 4g?

Reply:
looks like a typo. meant to use `<` instead of `>`.
```suggestion
    /* Don't allow threshold to be set above or even slightly below 4GB */
    if (sz > (1ull<<32) - 1<<20) {
```

Reply:
took a second look, i was wrong, there's no typo and `>` is ok.
the threshold must be below 4gb (the technical limit), so we error if it's greater.
updated my suggested patch above.

Reply:
Ohh, I did get confused.

Reply:
@perryitay please take the comment i suggested too